### ISSUE TITLE
Allow flag posts to be filtered

### ISSUE THREAD

**Issue by qlados** [PGMDev/PGM#543]
Currently a flag can have multiple posts, but you cant disable a certain by filtering it.

Code example:
```
<flags>
    <flag id="flag" name="Flag" post="post" points-rate="1" shared="true" pickup-kit="flag-kit" drop-kit="reset-health" sequential="true">
        <post id="mid" name="Mid" return-time="0s" respawn-time="20s" filter="only-first-spawn">
            <block>-351.5,12,510.5</block>
        </post>
        <post id="post-a" name="Post A" return-time="0s" respawn-time="20s">
            <block>-375.5,12,510.5</block>
        </post>
        <post id="post-b" name="Post B" return-time="0s" respawn-time="20s">
            <block>-328.5,12,510.5</block>
        </post>		
    </flag>
</flags>
<filters>
	<not id="only-first-spawn">
		<time>1s</time>
	</not>
</filters>
```

In this example the flag would be in mid at first, but never respawn back there because of the time filter. Instead it would just alternate between Post A and Post B.

**Comment by qlados** [comment]
Pablete said in Discord: "This is prone to breaking, what if no post passes the filter?"

Possible solutions:
- force atleast one unfiltered post
- loop the respawn countdown until a filter is matched
- force respawn the flag onto a post that is specified as default in the XML

**Comment by Pablete1234** [comment]
Resolved by #984
